alvr/dashboard/src/steamvr_launcher/linux_steamvr.rs

```diff
@@ -76,59 +76,124 @@ pub fn linux_hardware_checks() {
+
+    let have_nvidia_dgpu = device_infos.iter().any(|gpu| gpu.1 == DeviceInfo::Nvidia);
     debug!("have_nvidia_dgpu: {}", have_nvidia_dgpu);
+
+    let have_amd_igpu = device_infos.iter().any(|gpu| {
+        gpu.1
+            == DeviceInfo::Amd {
+                device_type: wgpu::DeviceType::IntegratedGpu,
+            }
+    });
+    debug!("have_amd_igpu: {}", have_amd_igpu);
+
     let have_amd_dgpu = device_infos.iter().any(|gpu| {
-        gpu == &DeviceInfo::Amd {
-            device_type: wgpu::DeviceType::DiscreteGpu,
-        }
+        gpu.1
+            == DeviceInfo::Amd {
+                device_type: wgpu::DeviceType::DiscreteGpu,
+            }
     });
     debug!("have_amd_dgpu: {}", have_amd_dgpu);
-    let have_intel_dgpu = device_infos.iter().any(|gpu| {
-        gpu == &DeviceInfo::Intel {
-            device_type: wgpu::DeviceType::DiscreteGpu,
+
+    if have_amd_igpu || have_amd_dgpu {
+        let is_any_amd_driver_invalid = device_infos.iter().any(|gpu| {
+            info!("Driver name: {}", gpu.0.get_info().driver);
+            match gpu.0.get_info().driver.as_str() {
+                "AMD proprietary driver" | "AMD open-source driver" => true, // AMDGPU-Pro | AMDVLK
+                _ => false,
+            }
+        });
+        if is_any_amd_driver_invalid {
+            error!("Amdvlk or amdgpu-pro vulkan drivers detected, SteamVR may not function properly. \
+            Please remove them or make them unavailable for SteamVR and games you're trying to launch. \
+            For more detailed info visit wiki: https://github.com/alvr-org/ALVR/wiki/Linux-Troubleshooting#artifacting-no-steamvr-overlay-or-graphical-glitches-in-streaming-view")
         }
+    }
+
+    let have_intel_dgpu = device_infos.iter().any(|gpu| {
+        gpu.1
+            == DeviceInfo::Intel {
+                device_type: wgpu::DeviceType::DiscreteGpu,
+            }
     });
     debug!("have_intel_dgpu: {}", have_intel_dgpu);
+
+    let vrmonitor_path_string = alvr_server_io::steamvr_root_dir()
+        .unwrap()
+        .join("bin")
+        .join("vrmonitor.sh")
+        .into_os_string()
+        .into_string()
+        .unwrap();
+    debug!("vrmonitor_path: {}", vrmonitor_path_string);
+
+    let mut vrmonitor_path_written = false;
     if have_igpu {
         if have_nvidia_dgpu {
-            warn!("For functioning VR you need insert following into SteamVR and ALL (!) games commandline options:");
+            warn!(
+                "For functioning VR you need to put following line into SteamVR commandline options and restart it:"
+            );
             warn!("__GLX_VENDOR_LIBRARY_NAME=nvidia __NV_PRIME_RENDER_OFFLOAD=1 __VK_LAYER_NV_optimus=NVIDIA_only \
-            VK_ICD_FILENAMES=/usr/share/vulkan/icd.d/nvidia_icd.json %command%")
+            VK_ICD_FILENAMES=/usr/share/vulkan/icd.d/nvidia_icd.json {} %command%", vrmonitor_path_string);
```

Comment:
I'm pretty sure `__VK_LAYER_NV_optimus=NVIDIA_only` and `VK_ICD_FILENAMES=/usr/share/vulkan/icd.d/nvidia_icd.json` do the same thing in two different ways. I'd rather not have it twice if not actually necessary as it ends to rather long copy-pastes.

Reply:
You could also extract the whole offload string into a separate variable if you're at it.

Reply:
hm... should we leave it like that? i forgot to change this, i mostly left it as is

Reply:
`VK_ICD_FILENAMES` would probably be the better way to fix this, only issue is when on some stupid distro that path doesn't match. And then we can just remove the vulkan layer. But we've already been emitting it like that and I've heard no real complaintd, so the path is probably fine.

Reply:
Or rather a long while ago `VK_DRIVER_FILES` was introduced as a replacement for the icd env var, so we could also use that instead.